accepted/2023/NuGetAudit-supress-advisory.md

```diff
@@ -0,0 +1,230 @@
+
+```xml
+<NuGetAuditSuppress>
+  https://example.com/CVE1;
+  https://example.com/CVE2;
+<NuGetAuditSuppress>
+```
+
+While creating a new MSBuild item for each advisory URL uses more characters I feel it "fits" better with XML & MSBuild generally.
+
+Finally, MSBuild properties don't allow additional metadata.
+By using an item, we can add metadata, such as `Packages`: `<NuGetAuditSuppress Include="https://example.com/CVE1" Packages="ExamplePackage1;ExamplePackage2" />`.
+For package sources this would have been helpful to enable credentials to authenticated package sources.
+For `NuGetAuditSuppress`, it could enable new future features more easily.
+
+### Metadata on PackageReference items
+
+Consider metadata on a `PackageReference` item:
+
+```xml
+<PackageReference
+    Include="SomePackageId"
+    Version="1.2.3"
+    NuGetAuditSuppress="https://sample.test/CVE1;https://sample.test/CVE2" />
+```
+
+When a single advisory is relevant to multiple packages, as is common for .NET, ASP.NET, and NuGet advisories, it will require duplication in multiple packages.
+
+Additionally, this will only work for direct packages, not transitive packages.
+If we use this approach for direct packages, there's no obvious alternative for transitive packages.
+Using a new MSBuild item for transitive packages, but metadata on a PackageReference item for direct packages is confusing because developers need more context in order to choose the correct syntax.
+It also means duplication when a package is direct in one project, but transitive in others.
+Having the suppress metadata flowed down to transitive packages, as Include/Exclude/Private Assets does is another option, because it's behavior that already exists.
+However, this makes it difficult to suppress centrally, like in a *Directory.Packages.props* file, as a transitive package will have different "entry points" in different projects in a solution.
+
+### CollectNuGetAuditSuppress target
+
+We already have `CollectPackageReferences`, `CollectPackageDownloads`, `CollectFrameworkReferences`, and `CollectCentralPackageVersions`.
+I believe the original design of these targets was specifically for `PackageDownload`, to allow the .NET SDK to determine instruct NuGet to download runtime packs and BCL reference assemblies only when they're not already available locally.
+I'm not sure why the other `Collect*` targets were added.
+This has a side-effect that it provides customers, and other MSBuild SDK authors, a way to customize their build and dynamically add or modify items as needed.
+
+We will add a `CollectNuGetAuditSuppress` target with the new `NuGetAuditSuppress` item. We will need to not only modify our own *NuGet.targets* to add it to the appropriate `DependsOnTargets`, but we also need to investigate MSBuild's static restore or .NET SDK to see if it needs to be added to the list of targets it "hardcodes" it runs.
```

Comment:
Do we absolutely need this target? 
Can we avoid adding a target for this? Something we should check with the CPS/project-system teams, whether there's a technical need for this target (guessing no). 

At this point, I don't see why we need an extensibility over this.

Reply:
> I spoke to Tom Meschter, and he confirmed that we do need a target if we want to collect any items and pass them onto the VS restore path.

https://github.com/NuGet/Home/pull/12921#discussion_r1508432638